### PR TITLE
bhead transition is now tick, w/o header check

### LINKED ISSUE #1052: Shelley Chain Tick
Implement the Chain tick function in shelley as described in #1046

### PR DESCRIPTION
**Pull request by JaredCorduan**
The `BHEAD` transition is now called `TICK`, and the actual header checks are moved outside to the `CHAIN` transition. `TICK` now just has a slot for a signal.

closes #1052

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/src/STS/Chain.hs [hunk 1]
```diff
@@ -67,8 +68,10 @@ instance
   type Environment (CHAIN crypto) = Slot
 
   data PredicateFailure (CHAIN crypto)
-    = BbodyFailure (PredicateFailure (BBODY crypto))
-    | BheadFailure (PredicateFailure (BHEAD crypto))
+    = HeaderSizeTooLargeTICK
```
Comment (nc6) [inline shelley/chain-and-ledger/executable-spec/src/STS/Chain.hs]: Why the postfix "TICK" here, given these are no longer in the `TICK` rule

File: shelley/chain-and-ledger/formal-spec/chain.tex [hunk 2]
```diff
@@ -675,41 +675,25 @@ \subsection{Reward Update Transition}
   \label{fig:rules:reward-update}
 \end{figure}
 
-\subsection{Block Header Transition}
-\label{sec:block-header-trans}
+\subsection{Chain Tick Transition}
+\label{sec:tick-trans}
 
-The Block Header Transition checks a couple of sizes and performs some chain level
+The Chain Tick Transition checks a couple of sizes and performs some chain level
```
Comment (nc6) [inline shelley/chain-and-ledger/formal-spec/chain.tex]:
```suggestion
The Chain Tick Transition performs some chain level
```

File: shelley/chain-and-ledger/formal-spec/chain.tex [hunk 3]
```diff
@@ -675,41 +675,25 @@ \subsection{Reward Update Transition}
...
-  protocol parameters allow for block headers.
-\item The size of the block body, as claimed by the block header, is less than or equal to the
-  maximal size that the protocol parameters allow for block bodies.
-  It will later be verified that the size of the block body matches the size claimed
-  in the header (see Figure~\ref{fig:rules:bbody}).
-\end{itemize}
+The $\mathsf{TICK}$ transition rule is shown in Figure~\ref{fig:rules:tick}.
+The signal is a slot \var{s}.
 
 If the size checks pass, then three transitions are done:
```
Comment (nc6) [inline shelley/chain-and-ledger/formal-spec/chain.tex]:
```suggestion
Three transitions are done:
```

File: shelley/chain-and-ledger/executable-spec/src/STS/Chain.hs (hunk 1 above)
Comment (JaredCorduan) [reply shelley/chain-and-ledger/executable-spec/src/STS/Chain.hs]: it shouldn't be, good catch!

**Pull request merged**